panos_upgrade_assurance/snapshot_compare.py

```diff
@@ -693,10 +701,12 @@ def get_count_change_percentage(
             passed=True,
         )
 
-        if self.left_snap[report_type].keys() != self.right_snap[report_type].keys():
+        requested_elements = set(next(iter(unary_dict)) for unary_dict in thresholds)
```

Comment:
why not simply `thresholds.keys()` ?

Reply:
`thresholds` is a list of single key-value dictionaries. So I can't get the list of keys of each dict within the list with `keys()` method.

```python
"thresholds": [
    {"num-max": 10},
    {"num-tcp": 10},
    {"num-pfcpc": 10}
]
```

Reply:
I'm wondering why we did a list here, not a dict 🤔 it would be more natural with a dict...

you're right of course